packages/nextjs/src/common/wrapGenerationFunctionWithSentry.ts

```diff
@@ -44,14 +45,23 @@ export function wrapGenerationFunctionWithSentry<F extends (...args: any[]) => a
       }
 
       const isolationScope = commonObjectToIsolationScope(headers);
+      let pathname = undefined as string | undefined;
 
       const activeSpan = getActiveSpan();
       if (activeSpan) {
         const rootSpan = getRootSpan(activeSpan);
         const { scope } = getCapturedScopesOnSpan(rootSpan);
         setCapturedScopesOnSpan(rootSpan, scope ?? new Scope(), isolationScope);
+
+        const spanData = spanToJSON(rootSpan);
+
+        if (spanData.data && 'http.target' in spanData.data) {
```

Comment:
l: this can be simplified to:

```js
if (typeof spanData.data['http.target'] === 'string') {
  pathname = spanData.data['http.target'];
}
```

`spanData.data` should never be empty I believe, and we can just check for stringiness and then avoid the typecase below :)